### ISSUE TITLE
[Messenger] Change doctrine middleware order 

### ISSUE THREAD

**Issue by wtorsi** [symfony/symfony-docs#11868]
Please, change doctrine middleware order in docs [https://symfony.com/doc/current/messenger.html#middleware-for-doctrine](https://symfony.com/doc/current/messenger.html#middleware-for-doctrine) 
The valid order for full stack of doctrine middleware is: (also, please include default_bus option)
```yaml
framework:
    messenger:
        default_bus: command.bus
        buses:
            command.bus:
                middleware:
                    - doctrine_ping_connection
                    - doctrine_close_connection
                    - doctrine_transaction
```

**Comment by weaverryan** [comment]
@wtorsi Can you tell us why this is the correct order? We're not necessarily experts on this, so having some validation behind this will make it much easier to do :).

Cheers!

**Comment by wtorsi** [comment]
Sure,
Middleware in messenger has stack behaviour, it means that top middleware calls next (below) middleware. 
In our case:
`DoctrinePingConnectionMiddleware` previously checks connection and reopen it if needed, then it calls next middleware in the stack.
`DoctrineCloseConnectionMiddleware` previously calls next middleware in the stack, and then it closes the connection. 
`DoctrineTransactionMiddleware` firstly starts the transaction, secondly calls next middleware in the stack (usually it is `HandleMessageMiddleware` which doing anything we want in our custom handler), and finally it flushes changes and commits the transaction.

The right business logic should be: `check connection` -> `start transaction, do things, commit` -> `close connection`.
According to the above short middleware description, the middleware stack should be
1. `DoctrinePingConnectionMiddleware` - it will check connection, then it will call next middleware `DoctrineCloseConnectionMiddleware`
2. `DoctrineCloseConnectionMiddleware`- it will call next middleware (wrap)  `DoctrineTransactionMiddleware`, and then will close the connection
3. `DoctrineTransactionMiddleware` - will be called inside `DoctrineCloseConnectionMiddleware`, will do anything we want in one transaction.

**Comment by OskarStark** [comment]
@weaverryan can you please confirm this?

**Comment by javiereguiluz** [comment]
@Nyholm could you please check if the proposed change is correct? Thanks!

**Comment by Nyholm** [comment]
Well spotted @wtorsi. You are very much correct. We should update the order. 

Do you want to make a PR to fix this?

**Comment by xabbuh** [comment]
fixed in #12357
